server.go

```diff
@@ -2513,6 +2526,7 @@ func newServer(listenAddrs []string, db database.DB, chainParams *chaincfg.Param
 		},
 		AddrIndex:       s.addrIndex,
 		ExistsAddrIndex: s.existsAddrIndex,
+		RegenTemplate:   func() { s.bgBlkTmplGenerator.RegenTemplate() },
```

Comment:
This will crash when `cfg.miningAddrs` is 0 because it's not set to anything in that case.

```
func() {
	if s.bgBlkTmplGenerator != nil {
		s.bgBlkTmplGenerator.RegenTemplate()
	}
},
```